### DISCUSSION TITLE
Adding KE and HalfLife to a 1 cmt model

### DISCUSSION THREAD

**Discussion by dan-adi** [metrumresearchgroup/mrgsolve#1026] [Q&A] [answered]
I am learning how to use R and mrgsolve and I was trying to add KE and drug half life to the model. 
This is what I've tried in the `pk1cmt` example file:

```$GLOBAL
#define CP (CENT/VC)
#define CLNL (VMAX/(KM+CP))
#define KE (CL/VC)
#define T12 (log(2) * VC/CL)
```
and
```
$MAIN
double VC = exp(log(TVC) + log(WT/70));
double KE = CL/VC;
double T12 = log(2) * VC/CL;
```
and
```
$CAPTURE @annotated
CP : Plasma concentration (mass/volume)
VC : Volume of distribution (volume)
KE : Elimination rate constant
T12: Half life
```
When I try to ``mread("pk1cmt_Ceftazidime")`` I get the following error:
```
12:4: error: expression is not assignable
KE = CL/VC;
~~ ^
13:5: error: expression is not assignable
T12 = log(2) * VC/CL;
~~~ ^
2 errors generated.
make: *** [pk_1_ceftamil-mread-source.o] Error 1
```
Any ideas on where I should compute those params?
Thank you

**Comment by dan-adi** [comment]
I've looked on some examples of code, namely popmodel.cpp, and I found this way of doing it:
```
$TABLE
double T12 = (log(2) * VC/CL);
double KE = CL/VC;
```
After this the values are computed and I am able to capture them like this:
```
out <- model_type2 %>%
  ev(e) %>% 
  Req(CP,T12,KE) %>%
  mrgsim(end = 24, delta = 0.1) 
```
Hope this is the correct way of doing it

**Comment by kylebaron** [comment, marked as answer]
Hi @dan-adi  - 

Thanks for giving R and mrgsolve a try.

That last example you posted looks right. Anything that you want to get out of the model but not needed to advance the system can be defined in `$TABLE` (or `$ERROR` ... same thing) and then captured. 

LMK if we can help with any other questions. 

Kyle

**Reply by dan-adi** [reply]
Thanks, I'm sure I'll have more questions down the road
